### PR TITLE
Add CDSAPI_KEY to workflow

### PR DESCRIPTION
**Pull request by yawenzzzz**

### CODE REVIEW

File: .github/workflows/data-source-integration-test.yml [hunk 1]
```diff
@@ -105,6 +105,7 @@ jobs:
             -e TEST_USGS_LANDSAT_USERNAME="${{ secrets.TEST_USGS_LANDSAT_USERNAME }}" \
             -e TEST_USGS_LANDSAT_PASSWORD="${{ secrets.TEST_USGS_LANDSAT_PASSWORD }}" \
             -e PL_API_KEY="${{ secrets.PL_API_KEY }}" \
+            -e CDSAPI_KEY="${{ secrets.CDSAPI_KEY }}" \
```
Comment (Hgherzog) [inline .github/workflows/data-source-integration-test.yml:108]: Can you trigger a data source integration workflow run and link it on this pr
?

**Review by Hgherzog — APPROVED** [review]
approved, assuming test passes

File: .github/workflows/data-source-integration-test.yml (hunk 1 above)
Comment (yawenzzzz) [reply .github/workflows/data-source-integration-test.yml:108]: Sure, https://github.com/allenai/rslearn/actions/runs/12286357443

Comment (yawenzzzz) [reply .github/workflows/data-source-integration-test.yml:108]: alright, the new run passed, I will merge the change.

**Pull request merged**